SPECIFICATION.md

```diff
@@ -0,0 +1,93 @@
+<h1>Provider Specification</h1>
+
+[TOC]
+
+# General
+- **name** Any record name supplied to the provider must be the full hostname.
+- **ttl** Reasonable default is 6 hours since it's supported by most services. Any service that does not support this must be explicitly mentioned somewhere.
+
```

Comment:
We should add a general definition for a lexicon record here:

- **record** All provider/API records will be translated into a common lexicon record format. This format has the following structure:

```
{
    'type': string, // upper case, valid record type. eg. A, CNAME, TXT
    'name': string, // lowercase, full domain name. eg. test.record.example.com
    'ttl': integer, // positive integer, in seconds. eg. 3600
    'content': string,
    'id': string // optional, provider specified unique id. 
}
```